### PR TITLE
add dns ttl to dns dissector and ndpi dns proto struct

### PR DESCRIPTION
**Pull request by hl33ta**
See above :)
This MR just adds the ttls field as part of the ndpi dns dissector and the dns portion of the ndpi flow structure

### CODE REVIEW

File: src/lib/protocols/dns.c [hunk 1]
```diff
@@ -140,6 +140,16 @@ static u_int16_t get16(int *i, const u_int8_t *payload) {
 
 /* *********************************************** */
 
+static u_int32_t get32(int *i, const u_int8_t *payload) {
```
Comment (spendletonliveaction) [inline src/lib/protocols/dns.c:143]: Is this called get32 because there is a get16 for 16 bit ints?

Comment (hl33ta) [reply src/lib/protocols/dns.c:143]: yup

File: src/lib/protocols/dns.c [hunk 2]
```diff
@@ -277,6 +287,7 @@ static int search_valid_dns(struct ndpi_detection_module_struct *ndpi_struct,
 
 	  rsp_type = get16(&x, packet->payload);
 
+
```
Comment (shla2022) [inline src/lib/protocols/dns.c]:
```suggestion
```

File: src/lib/protocols/dns.c [hunk 3]
```diff
@@ -287,7 +298,9 @@ static int search_valid_dns(struct ndpi_detection_module_struct *ndpi_struct,
 
 	  /* here x points to the response "class" field */
 	  if((x+12) <= packet->payload_packet_len) {
-	    x += 6;
+      x += 2;
```
Comment (shla2022) [inline src/lib/protocols/dns.c]: Suggest ensuring whitespace matches upstream.

**Comment by shla2022** [comment]
Minor stylistic comments only.

**Review by shla2022 — APPROVED** [review]

File: src/lib/protocols/dns.c (hunk 3 above)
Comment (hl33ta) [reply src/lib/protocols/dns.c]: done in recent push

**Pull request merged**